src/ax/saga/mapping/keyset/mapper.go

```diff
@@ -0,0 +1,65 @@
+package keyset
+
+import (
+	"context"
+
+	"github.com/jmalloc/ax/src/ax"
+	"github.com/jmalloc/ax/src/ax/persistence"
+	"github.com/jmalloc/ax/src/ax/saga"
+)
+
+// Mapper is an implementation of saga.Mapper that maps messages to saga
+// using disjoint "key sets".
+//
+// This is a flexible mapping strategy that allows the saga precise control over
+// which instance to load on a per-message basis.
+//
+// The saga must implement the keyset.Saga interface to work with this mapper.
+type Mapper struct {
+	Repository Repository
+}
+
+// MapMessageToInstance returns the ID of the saga instance that is the target
+// of the given message.
+//
+// If no existing saga instance is found, it returns false.
```

Comment:
As with the interface, this does not return `false`. It should be `MapResultNotFound`, or just say that it returns a MapResult.